### DISCUSSION TITLE
Right-to-left mode discussion

### DISCUSSION THREAD

**Discussion by MarcinZiabek** [QuestPDF/QuestPDF#256] [Ideas]
The next QuestPDF iteration will introduce the RTL mode. Let's discuss details!

**Comment by aealduraiweesh** [comment]
Hi im a native Arabic speaker. Let me know if i can help verify anything

**Reply by MarcinZiabek** [reply]
Hi 😁 Your help will be highly appreciated now. I made first progress regarding proper line breaking in Arabic. Would you like to take a look in https://github.com/QuestPDF/QuestPDF/pull/349?

**Reply by NaseemHajoj** [reply]
Hi Marcin  I think you're doing great job with this library, waaay better and easier to write a PDF with your library than competitors.
I can also help in verifying both Arabic and Hebrew if this is still needed. And in coding stuff too.
Keep up the great job

**Comment by lmingle** [comment]
Another thought to go along with the RTL discussion is how it could be
expanded to eventually lead to including Tategaki. This could define
the document layout as .Columns or .Lines as the default.

From wikipedia - tategaki (縦書き), which was inherited from traditional
Chinese practice. In this format, the characters are written in *columns
going from top to bottom, with columns ordered from right to left*.

**Reply by MarcinZiabek** [reply]
How popular is this writing system? Text-related capabilities are surprisingly complex, taking a lot of time to implement, and introduce high risk of mistakes. I would like to focus on the most useful features first 😁

**Comment by MarcinZiabek** [comment]
The development of this effort has been started in the https://github.com/QuestPDF/QuestPDF/pull/349 😁

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
